torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -555,6 +862,14 @@ void CompiledFusionFunction::launch_with_tensors(at::ArrayRef<at::Tensor> inputs
       }
     }
   }
+  // If the kernel call contains a random op, we need to pass in random seeds as
+  // well.
+  if(has_random) {
+    auto gen_ = THCRandom_getGenerator(at::globalContext().getTHCState());
+    uint64_t offset = gen_->state.philox_seed_offset.fetch_add(20);
```

Comment:
20? That seems... very questionable :)

Reply:
At compilation time, don't we know how many samples we need to take? So you can just save it in the compilation unit and then grab exactly the correct offset.

Reply:
Actually I wasn't quite sure what numbers to put here. I was looking at some previous usage of philox seed in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Distributions.cu and I just saw those random numbers are being used here.